### DISCUSSION TITLE
The command 'docker' could not be found in this WSL 2 distro.

### DISCUSSION THREAD

**Discussion by Prathipgi** [microsoft/WSL#9951] [General]
Unable to connect Docker with WSL

The command 'docker' could not be found in this WSL 2 distro.
We recommend to activate the WSL integration in Docker Desktop settings.

For details about using Docker Desktop with WSL 2, visit:

https://docs.docker.com/go/wsl2/
![error](https://user-images.githubusercontent.com/130360426/231284273-4460f17c-283a-4795-882d-b7fa810430aa.PNG)

**Comment by Biswa96** [comment]
It's working fine in my system.

![image](https://user-images.githubusercontent.com/31443074/231334869-30235aa2-8382-415c-9b45-8500f7871d70.png)

Did you install docker with this documentation https://docs.docker.com/engine/install/ubuntu/ ?

**Reply by Prathipgi** [reply]
Yes

**Comment by Phascolarctos** [comment]
### check your hyper-v enable in bios and virtual machine platform enabled in programs
### check your wsl version on 2，you must cost some time on install docker on version 1
### 
`
wsl --status
`
`
wsl -l -v
`
`
wsl --set-version Ubuntu-22.04 2
`
`
service docker start
`

**Reply by Alena009** [reply]
You saved my day! +100500 to you

**Comment by AnaMaghradze** [comment]
For me, uninstall and then install worked. Just followed these steps:
https://learn.microsoft.com/en-us/virtualization/windowscontainers/manage-docker/configure-docker-daemon#prepare-your-system-for-dockers-removal
